src/lib/marquee.tsx

```diff
@@ -1,23 +1,40 @@
-import PropTypes from "prop-types"
-import { useState, useMemo, useRef, useEffect, useCallback } from "react"
+import { useState, useMemo, useRef, useEffect, useCallback, ReactElement } from "react"
 
 import { classnames } from "./utils/classnames"
 
 import "./marquee.scss"
 
-/**
- * Infinite scrolling marquee component.
- *
- * @param {object} props
- * @param {number=50} props.speed - The speed of the marquee in pixels per second.
- * @param {number=20} props.reducedMotionSpeed - The speed of the marquee in pixels per second when prefersReducedMotion is true.
- * @param {boolean=false} props.prefersReducedMotion - If true, marquee will use reducedMotionSpeed instead of speed.
- * @param {boolean=true} props.playing - If false, the marquee will not animate.
- * @param {string} props.className - Additional classnames to add to the marquee.
- * @param {React.Component} props.children - The content to be rendered inside the marquee. Will be wrapped in a div with the class of `marquee__slide`.
- *
- * @returns {React.Component}
- */
+export interface MarqueeProps extends React.Component {
```

Comment:
This is bit more forgiving as it allows users of the component to pass any valid HTML attribute that a `div` can have.

Important as you can then pass `data-` and `aria-` attributes as needed.

```suggestion
export interface MarqueeProps extends React.ComponentPropsWithoutRef<"div"> {
```